src/test/java/com/microsoft/graph/http/BaseCollectionPageTests.java

```diff
@@ -35,8 +35,8 @@ public void setUp() throws Exception {
         list.add("Object2");
         list.add("Object3");
         final IBaseClient<Request> mBaseClient = mock(IBaseClient.class);
-        mRequestBuilder = new BaseRequestBuilder<>(requestUrl, mBaseClient, null) {};
-        baseCollectionPage = new BaseCollectionPage<>(list, mRequestBuilder) {};
+        mRequestBuilder = new BaseRequestBuilder<String>(requestUrl, mBaseClient, null) {};
```

Comment:
can you please revert the changes on the generic types? thanks!

Reply:
java 8  cannot use '<>' with anonymous inner classes

Reply:
We're currently compiling with java 15, could we then change the compilation version to 15? FYI the pom.xml is here to github can map the dependency trees, but we actually use gradle.

Reply:
docs saying 

6. Supported Java versions
The Microsoft Graph SDK for Java library is supported at runtime for Java 8 and Android API revision 26 or greater.

should it compiled fine on Java 8 ?

Reply:
as far as I understand the compile java version and the runtime java version are two completely different things. You can perfectly have something being compiled by java 15 but target a runtime of java 8. And the compilation with java 15 brings a bunch of "productivity features" that are not available with the v8 compiler, which will be compiled to java 8 byte code. In fact that's exactly what we're doing. 
https://stackoverflow.com/questions/16654951/gradle-sourcecompatibility-vs-targetcompatibility